src/SDKs/Azure.Base/data-plane/Azure.Base/Http/Pipeline/ExponentialRetryPolicy.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+
+namespace Azure.Base.Http.Pipeline
+{
+    internal class ExponentialRetryPolicy : RetryPolicy
+    {
+        private readonly int[] _retriableCodes;
+
+        private readonly Func<Exception, bool> _exceptionFilter;
+
+        private readonly int _maxRetries;
+
+        private readonly TimeSpan _delay;
+
+        private readonly TimeSpan _maxDelay;
+
+        private Random _random;
+
+        public ExponentialRetryPolicy(int[] retriableCodes, Func<Exception, bool> exceptionFilter, int maxRetries, TimeSpan delay, TimeSpan maxDelay)
+        {
+            _random = new ThreadSafeRandom();
+            _exceptionFilter = exceptionFilter;
+            _maxRetries = maxRetries;
+            _delay = delay;
+            _maxDelay = maxDelay;
+
+            _retriableCodes = retriableCodes.ToArray();
+            Array.Sort(_retriableCodes);
+        }
+
+        protected override bool ShouldRetry(HttpPipelineMessage message, Exception exception, int attempted, out TimeSpan delay)
+        {
+            delay = TimeSpan.FromMilliseconds(
```

Comment:
nit:  As exponential retry backoffs are potentially unclear to many, it may be difficult to grok and maintain without some insight.  You may want to consider commenting on the approach and the components of the calculation for clarity.  For illustration, I personally break it down similar to the following,:

```c#
        /// <summary>
        ///   Creates a retry policy with exponential backoff for use with external operations.
        /// </summary>
        /// 
        /// <param name="rng">The random number generator to use for calculating jitter in the retry period. </param>
        /// <param name="maxRetryAttempts">The maximum number of retry attempts before giving up.</param>
        /// <param name="exponentialBackoffSeconds">The number of seconds on which to base the exponential backoff.</param>
        /// <param name="baseJitterSeconds">The base number of seconds to use when including random jitter.</param>
        /// 
        /// <returns>The retry policy.</returns>
        /// 
        ... codez ...
```